cmd/notary/keys_test.go

```diff
@@ -806,3 +807,31 @@ func TestImportKeysNonexistentFile(t *testing.T) {
 	err = k.importKeys(&cobra.Command{}, []string{"Idontexist"})
 	require.Error(t, err)
 }
+
+func TestKeyGeneration(t *testing.T) {
+	tempDir := tempDirWithConfig(t, "{}")
+	defer os.RemoveAll(tempDir)
+
+	_, err := runCommand(t, tempDir, "key", "generate", data.ECDSAKey, "--role", "targets")
+	require.NoError(t, err)
+	assertNumKeys(t, tempDir, 0, 1, false)
+
+	_, err = runCommand(t, tempDir, "key", "generate", data.ECDSAKey, "--role", "targets", "-o", filepath.Join(tempDir, "testkeys"))
+	require.NoError(t, err)
+	assertNumKeys(t, tempDir, 0, 1, false) // key shouldn't be written to store and won't show up in keylist
+
+	// test that we can read the keys we created
```

Comment:
could we additionally test a `key rotate` and/or `key import` to check that the notary CLI can ingest the key directly?